src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import core.basesyntax.service.FruitConverter;
+import core.basesyntax.service.exceptions.InvalidDataException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DataConverterImpl implements DataConverter {
+    private static final int NUMBER_OF_COLUMNS = 3;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        return inputReport.stream()
+                .map(a -> a.trim())
+                .map(a -> a.split(","))
+                .filter(a -> isDataCorrect(a))
+                .map(el -> {
+                    if (isInteger(el[2].trim())) {
+                        FruitTransaction transaction = new FruitTransaction();
+                        transaction.setOperation(FruitTransaction.Operation.fromCode((el[0])));
+                        transaction.setFruit(FruitConverter.convertToFruit(el[1]));
+                        transaction.setQuantity(Integer.valueOf(el[2]));
+                        return transaction;
+                    } else {
+                        return null;
+                    }
+                })
+                .collect(Collectors.toList());
+    }
+
+    private boolean isDataCorrect(String[] line) {
+        if (line.length != NUMBER_OF_COLUMNS) {

```

Comment:
The isDataCorrect() method returns a boolean but always returns true or throws an exception. Consider changing the method to void and renaming it to validateData() or similar, as it doesn't really return a meaningful result.

Reply:
but I use this method in filter...